util/symbolize/addr2line_inlinestack.cc

```diff
@@ -378,35 +406,103 @@ void InlineStackHandler::ProcessAttributeUnsigned(
         break;
       case DW_AT_stmt_list:
         {
-          SectionMap::const_iterator iter =
-              sections_.find(".debug_line");
-          CHECK(iter != sections_.end()) << "unable to find .debug_line "
+          SectionMap::const_iterator line_sect = sections_.find(".debug_line");
+          CHECK(line_sect != sections_.end()) << "unable to find .debug_line "
               "in section map";
-          SectionMap::const_iterator line_str =
-              sections_.find(".debug_line_str");
+          SectionMap::const_iterator line_str = sections_.find(".debug_line_str");
           const char* line_str_buffer = NULL;
           uint64 line_str_size = 0;
           if (line_str != sections_.end()) {
             line_str_buffer = line_str->second.first;
             line_str_size = line_str->second.second;
           }
-          LineInfo lireader(iter->second.first + data,
-                            iter->second.second - data,
-                            line_str_buffer,
-                            line_str_size,
+          SectionMap::const_iterator str_section = sections_.find(".debug_str");
+          const char* str_buffer = NULL;
+          uint64 str_buffer_size = 0;
+          if (str_section != sections_.end()) {
+            str_buffer = line_str->second.first;
+            str_buffer_size = line_str->second.second;
+          }
+          SectionMap::const_iterator str_offset = sections_.find(".debug_line_str");
+          const char* str_offset_buffer = NULL;
+          uint64 str_offset_size = 0;
+          if (str_offset != sections_.end()) {
+            str_offset_buffer = line_str->second.first;
+            str_offset_size = line_str->second.second;
+          }                    
+          LineInfo lireader(line_sect->second.first + data, line_sect->second.second - data,
+                            line_str_buffer, line_str_size,
+                            str_buffer, str_buffer_size,
+                            str_offset_buffer, str_offset_size,
+                            get_str_offset_base(),
                             reader_, line_handler_);
+
           line_handler_->SetVaddrOfFirstLoadSegment(
               vaddr_of_first_load_segment_);
           lireader.Start();
           have_two_level_line_tables_ = lireader.have_two_level_line_tables();
         }
         break;
+      case DW_AT_str_offsets_base:
+        str_offset_base_ = data;
+        break;
+      case DW_AT_ranges: 
+        CHECK(form == DW_FORM_sec_offset); // fall
```

Comment:
Is this code expected to be executed only for DWARF5? If so, can we add an assert to make that explicit?

Reply:
Do we know if GCC always uses GNU extensions for `gsplit-dwarf` in `DWARF4`? It's possible that `DWARF4` support is not complete in autofdo and it would be good to know whether that's the case.

Reply:
Ok, sounds good.